### PR TITLE
Wi-Fi: Nrf wifi mgmt

### PR DESCRIPTION
**Pull request by krish2718**
This is based on #8495 will be rebased once its merged.

### CODE REVIEW

**Comment by greg-fer** [comment]
@krish2718 , remember to add tech writer to your PR when you're making doc changes. Thanks!

**Comment by krish2718** [comment]
@krga2022 this would change the CLI from `wpa_cli` -> `wifi`, so, please check README.rst and test.

**Review by richabp — COMMENTED** [review]
some minor suggestions.

**Comment by kkasperczyk-no** [comment]
Please update sdk-connectedhomeip revision with new SHA: d591ef25a13cc10d5d40937c323aa48244ed7b32

File: west.yml [hunk 1]
```diff
@@ -50,7 +50,7 @@ manifest:
     # https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/zephyr/guides/modules.html
     - name: zephyr
       repo-path: sdk-zephyr
-      revision: b341c1ffbee9e1694f07a62f55b76b3f4947f603
+      revision: pull/876/head
```
Comment (rlubos) [inline west.yml:53]: Zephyr revision: `e5ff5593aec822d2e478cae6a96cd7b00011aed2`

**Review by rlubos — APPROVED** [review]

**Comment by krish2718** [comment]
@Damian-Nordic @kkasperczyk-no any ideas about this failure https://jenkins-ncs.nordicsemi.no/job/latest/job/sub/job/test-fw-nrfconnect-chip/job/master/7029/console ? Seems to be environment/scripts related?

**Comment by krish2718** [comment]
https://jenkins-ncs.nordicsemi.no/job/latest/job/sub/job/test-fw-nrfconnect-tfm/job/master/6059/console this seems to be a HW failure (`An error was reported by NRFJPROG DLL: -10 EMULATOR_NOT_CONNECTED`)

**Review by richabp — APPROVED** [review]

**Comment by rlubos** [comment]
The single CI failure is unrelated (and was passing in previous runs), merging

**Pull request merged**